api/src/ap/java/com/velocitypowered/api/plugin/ap/SerializedPluginDescription.java

```diff
@@ -14,14 +14,15 @@
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.List;
+import java.util.Locale;
 import java.util.Objects;
 import java.util.regex.Pattern;
 import java.util.stream.Collectors;
 import org.checkerframework.checker.nullness.qual.Nullable;
 
 public final class SerializedPluginDescription {
 
-  public static final Pattern ID_PATTERN = Pattern.compile("[a-z][a-z0-9-_]{0,63}");
+  public static final Pattern ID_PATTERN = Pattern.compile("[a-z-A-Z][a-z-A-Z-0-9-_]{0,63}");
```

Comment:
This regex contains extra dashes between the z and the A (e.g. "-foo" is incorrectly accepted as a valid ID).